crates/matrix-sdk/src/client/mod.rs

```diff
@@ -1305,31 +1322,64 @@ impl Client {
         ))
         .await;
 
-        // If this is an `M_UNKNOWN_TOKEN` error and refresh token handling is active,
-        // try to refresh the token and retry the request.
-        if self.inner.handle_refresh_tokens {
-            if let Err(Some(ErrorKind::UnknownToken { .. })) =
-                res.as_ref().map_err(HttpError::client_api_error_kind)
-            {
-                if let Err(refresh_error) = self.refresh_access_token().await {
-                    match &refresh_error {
-                        RefreshTokenError::RefreshTokenRequired => {
-                            // Refreshing access tokens is not supported by
-                            // this `Session`, ignore.
-                        }
-                        _ => {
-                            return Err(refresh_error.into());
-                        }
+        // An `M_UNKNOWN_TOKEN` error can potentially be fixed with a token refresh.
```

Comment:
I feels very weird to me that we have this logic duplicated here and in the `futures.rs`, again, currently just building on what exists right now.

Reply:
Would be super nice to deduplicate, though, and it seems to be relatively minimal effort. Or it could be done in a follow-up PR.

Reply:
Lets take this in a follow-up PR 👍